engine/runtime/src/main/java/org/enso/interpreter/runtime/data/vector/ArrayBuilder.java

```diff
@@ -68,6 +68,23 @@ void add(Object e, WarningsLibrary warnings) {
         longArray[size++] = l;
       } else {
         CompilerDirectives.transferToInterpreter();
+        TRY_DOUBLE:
```

Comment:
- The change is getting complicated
- after [problems with propagation of warnings](https://github.com/enso-org/enso/runs/42526347007#r2s286) in a `Vector.Builder`: 
   - Builder should not propagate warnings attached to an item `with .append_vector_range with propagate_warnings=False`
	- `[Error err] did not equal []; lengths differ (1 != 0)  (at Base_Tests/src/Data/Vector_Spec.enso:1220:13-47).`
- I decided to write `VectorBuilderTest` to check the warnings
- but it turns out we don't create effective backing for mixed `long` and `double` values
   - that may affect us badly and mixed values like that may be pretty common
   - hence fixes like 8fb3cbc and b7eb0a2